webportal/src/app/cluster-view/services/service-info.js

```diff
@@ -0,0 +1,43 @@
+// Copyright (c) Microsoft Corporation
+// All rights reserved.
+//
+// MIT License
+//
+// Permission is hereby granted, free of charge, to any person obtaining a copy of this software and associated
+// documentation files (the "Software"), to deal in the Software without restriction, including without limitation
+// the rights to use, copy, modify, merge, publish, distribute, sublicense, and/or sell copies of the Software, and
+// to permit persons to whom the Software is furnished to do so, subject to the following conditions:
+// The above copyright notice and this permission notice shall be included in all copies or substantial portions of the Software.
+//
+// THE SOFTWARE IS PROVIDED *AS IS*, WITHOUT WARRANTY OF ANY KIND, EXPRESS OR IMPLIED, INCLUDING
+// BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY, FITNESS FOR A PARTICULAR PURPOSE AND
+// NONINFRINGEMENT. IN NO EVENT SHALL THE AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM,
+// DAMAGES OR OTHER LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+// OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE SOFTWARE.
+
+// This function will call kubernetes restful api to get node - podlist - label info, to support service view monitor page.
+function getServiceView(kubernetesUrl) {
+  var nodesInfo = []
+  var request = require('sync-request');
+  var resNodes = request('GET', kubernetesUrl + '/api/v1/nodes');
+  var data = JSON.parse(resNodes.body.toString('utf-8'));
+  var items = data.items;
+  for (var i in items) {
```

Comment:
Hi Yanjie, you could look at this API: http://10.151.40.179:8080/api/v1/namespaces/default/pods/

It can get the all pods of a specified namespace. If we run all pods under default namespace, we can get the all pods with a single call.

In this call, we can know each pod detailed info and it nodeName. So loop through the results, and with the prepared nodeName->nodeInfo hash table, I think we can construct a full nodeList and their pods with just 2 ajax calls.

Reply:
it looks to me that we have come across this api in the meeting. we should put it in a meeting minutes next time before we forget

---
In reply to: [159404736](https://github.com/Microsoft/pai/pull/145#discussion_r159404736) [](ancestors = 159404736)

Reply:
Sorry to check this PR late, yes it covered in the meeting but not reflected in the code.

---
In reply to: [159406680](https://github.com/Microsoft/pai/pull/145#discussion_r159406680) [](ancestors = 159406680,159404736)